src/opnsense/www/js/opnsense.js

```diff
@@ -142,19 +147,20 @@ function setFormData(parent,data) {
                         });
                     } else if (targetNode.prop("type") == "checkbox") {
                         // checkbox type
-                        if (node[keypart] != 0) {
-                            targetNode.prop("checked",true);
-                        } else {
-                            targetNode.prop("checked",false);
-                        }
+                        targetNode.prop("checked", node[keypart] != 0);
                     } else if (targetNode.is("span")) {
                         if (node[keypart] != null) {
                             targetNode.text("");
                             targetNode.append(htmlDecode(node[keypart]));
                         }
                     } else {
                         // regular input type
-                        targetNode.val(htmlDecode(node[keypart]));
+                        if (targetNode.hasClass('json-data')) {
+                            // if the input field is JSON data, serialize the data into the field
+                            targetNode.val(JSON.stringify(node[keypart]));
```

Comment:
better to use a data tag for this, should avoid json serialisation:

`targetNode.data('json', node[keypart]);
`

Reply:
data does not use a data attribute but a jQuery cached data stucture but it is better to handle anyway. The only difference is that I used 'data' as a name, since it is just a JavaScript structure and has theoretically nothing to do with JSON.